### DISCUSSION TITLE
Yolov5 paging file

### DISCUSSION THREAD

**Discussion by zeleny-ondrej** [ultralytics/yolov5#5584] [Q&A]
So after hours and hours spend installing and reinstalling everything like 5 times, i finally managed to intall the corect packages compatible with my system. I finnaly started making some progress but when i decided to retraint the model for my aplication (car classification) i was met wit the "small paging file size" error.  The only way to overcome this issue i had to change the batch size to 1. Is my system that bad that it can hangle bigger batch sizes ?

I5 6660K
I have GTX 960 4GB
16 GB ram

![image](https://user-images.githubusercontent.com/38864594/140875763-d824e05e-dcf1-4817-bb46-86d70628c740.png)

**Comment by glenn-jocher** [comment]
@Zelda000 maybe try increasing paging file size?
https://www.google.com/search?q=windows+increase+paging+file+size&oq=windows+increase+paging+file&aqs=chrome.0.0i512j69i57j0i22i30l6j0i390l2.2993j0j7&sourceid=chrome&ie=UTF-8

**Reply by zeleny-ondrej** [reply]
Oh, ups i forgot to mention, i have tried setting the paging file size to up to 30GB but with no result, Sometime im able to do batch size 2.

Im guessing that the fact im using pycharm (installed on C:\) with anaconda as interpreter (installed on D:\) should not be an issue :D

**Reply by glenn-jocher** [reply]
@Zelda000 hmm interesting. I can't help you myself as I'm not on windows, but here is our general guideline for CUDA OOM issues.

👋 Hello! Thanks for asking about **CUDA memory issues**. YOLOv5 🚀 can be trained on CPU, [single-GPU](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data), or [multi-GPU](https://github.com/ultralytics/yolov5/issues/475). When training on GPU it is important to keep your batch-size small enough that you do not use all of your GPU memory, otherwise you will see a CUDA Out Of Memory (OOM) Error and your training will crash. You can observe your CUDA memory utilization using either the `nvidia-smi` command or by viewing your console output:

<img width="800" alt="Screenshot 2021-05-28 at 12 19 51" src="https://user-images.githubusercontent.com/26833433/131122318-8141d1ce-13de-44bf-bbdc-487bad442ff0.png">

### CUDA Out of Memory Solutions

If you encounter a CUDA OOM error, the steps you can take to reduce your memory usage are:

- Reduce `--batch-size`
- Reduce `--img-size`
- Reduce model size, i.e. from YOLOv5x -> YOLOv5l -> YOLOv5m -> YOLOv5s
- Train with [multi-GPU](https://github.com/ultralytics/yolov5/issues/475) at the same `--batch-size`
- Upgrade your hardware to a larger GPU
- Train on free GPU backends with up to 16GB of CUDA memory: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

### AutoBatch

You can use [YOLOv5 AutoBatch](https://github.com/ultralytics/yolov5/pull/5092) (NEW) to find the best batch size for your training by passing `--batch-size -1`. AutoBatch will solve for a 90% CUDA memory-utilization batch-size given your training settings. AutoBatch is experimental, and only works for Single-GPU training. It may not work on all systems, and is not recommended for production use.

<img width="1746" alt="Screenshot 2021-11-06 at 12 31 10" src="https://user-images.githubusercontent.com/26833433/140608327-9c0c0baf-a0ee-4551-a4ff-5ee26b155c51.png">

Good luck and let us know if you have any other questions!
